### PR TITLE
New block: Glossary Term

### PR DESCRIPTION
**Pull request by Morpheu5**
Add these inside Concept pages (might add a dedicated page type in the future) and brace for content errors, because the API does not know about these.

* 04fcd62 Allow editing of glossary terms IDs and add the option to have them auto-generated
* 2cdb6a8 Merge remote-tracking branch 'origin/master' into feature/glossary-terms
* 9e9d618 New block: Glossary Term
* 2cdb6a8 Merge remote-tracking branch 'origin/master' into feature/glossary-terms
* 0e96f78 Change content block to simple textarea element for glossary terms

### CODE REVIEW

**Comment by Morpheu5** [comment]
Needs a key-like id field for indexing.

File: app/js/app/directives/react_classes/GlossaryTermBlock.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                var emptyExplanation = {
+                    type: "content",
+                    children: [],
+                    encoding: "markdown"
+                };
+                return (
+                    <Block type="glossaryTerm" blockTypeTitle="Glossary term" doc={this.props.doc} onChange={this.onDocChange}>
+                        <div className="small-3 columns">
+                            <div className="row">
+                                <input type="text" value={this.props.doc.value} onChange={this.onContentChange} placeholder="Glossary term" />
```
Comment (bh412) [inline app/js/app/directives/react_classes/GlossaryTermBlock.js]: I think we should change the entry method to be more consistent with the other blocks, e.g. have a placeholder here, and change the explanation box on line 72 to be like the exposition boxes on the question pages - as we won't want it to be possible to put accordions (or other glossary terms) inside the glossary terms

File: app/snippets/content_templates/glossaryTerm.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"encoding": "markdown",
+	"value": "_Enter term here_",
```
Comment (bh412) [inline app/snippets/content_templates/glossaryTerm.json]: This doesn't quite look right as a placeholder, e.g. it actually sets it as the glossary term if you leave it (see other comment for suggestions)

Comment (Morpheu5) [reply app/snippets/content_templates/glossaryTerm.json]: Done in 0e96f78624bfac65c3b441d04566275f5259580f.

File: app/js/app/directives/react_classes/GlossaryTermBlock.js (hunk 1 above)
Comment (Morpheu5) [reply app/js/app/directives/react_classes/GlossaryTermBlock.js]: Done in 0e96f78624bfac65c3b441d04566275f5259580f.

**Comment by bh412** [comment]
looks good post changes, worth checking everything aligns with what the content team want

**Pull request merged**